keps/sig-api-machinery/3156-http3/README.md

```diff
@@ -0,0 +1,228 @@
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+One of the main advantages of HTTP/3 is increased performance, specifically around fetching multiple
+objects simultaneously. With HTTP/2, any interruption (packet loss) in the TCP connection blocks all
+streams (Head of line blocking).
+
+In addition, HTTP/3 offers 0-RTT support, which means that subsequent connections can start up much
+faster by eliminating the TLS acknowledgement from the server when setting up the connection. This
+means the client can start requesting data much faster than with a full TLS negotiation.
+
+## Motivation
+
+Kubernetes clusters defaulted to HTTP/2 for the control plane communications to solve all the
+performance and reliability problems caused by the previous protocol HTTP/1.
+However, there are some notable exceptions:
+
+- [communication between apiserver and webhooks](https://github.com/kubernetes/kubernetes/pull/82090).
+- [communication that use SPDY or websockets](https://github.com/kubernetes/kubernetes/issues/7452)
+
+HTTP/3 solves some of the HTTP/2 problems, mainly the Head of Line Blocking problem because it uses
+QUIC RFC9000 for the transport layer, improving the performance and the resilience of the control
+plane.
+
+Despite the HTTP/3 protocol, as January 2022, is still an [Internet Draft](https://datatracker.ietf.org/doc/draft-ietf-quic-http/), it is already supported by 73% of running web browsers, and there is also a [golang implementation](https://github.com/lucas-clemente/quic-go) that is being used by known projects and companies.
+
+Having support for HTTP/3 will allow the project to have early feedback, and the possibility to influence it.
+
+### Goals
+
+- Support HTTP/3 on the apiserver wherever is possible
```

Comment:
Does this include where the API server acts as the client?

Reply:
I should expand and clarify, only  for serving,  the others are implicitly covered in non-goals

Reply:
Explicit non-goals would help make that super clear, I think (and leave room for a separate KEP about HTTP/3 for _those_ cases)